src/sync.d

```diff
@@ -2129,4 +2197,75 @@ final class SyncEngine
 			writeln("No pending remote changes - in sync");
 		}
 	}
+	
+	// Create a fake OneDrive response suitable for use with saveItem
+	JSONValue createFakeResponse(string path) {
+		import std.digest.sha;
+		// Generate a simulated JSON response which can be used
+		// At a minimum we need:
+		// 1. eTag
+		// 2. cTag
+		// 3. fileSystemInfo
+		// 4. file or folder. if file, hash of file
+		// 5. id
+		// 6. name
+		// 7. parent reference
+		
+		SysTime mtime = timeLastModified(path).toUTC();
+		
+		// real id / eTag / cTag are different format for personal / business account
+		auto sha1 = new SHA1Digest();
+		ubyte[] hash1 = sha1.digest(path);
+		
+		JSONValue fakeResponse;
+		
+		if (isDir(path)) {
+			// path is a directory
+			fakeResponse = [
```

Comment:
Indentation is a bit weird here.

Reply:
Indentation was deliberately different to ensure the correct fake JSON response was being generated